rs/ethereum/cketh/minter/src/eth_logs/parser.rs

```diff
@@ -164,22 +164,29 @@ impl LogParser for ReceivedEthOrErc20LogParser {
 fn ensure_not_pending(entry: &LogEntry) -> Result<(BlockNumber, EventSource), ReceivedEventError> {
     let _block_hash = entry
         .block_hash
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
     let block_number = entry
         .block_number
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
     let transaction_hash = entry
         .transaction_hash
+        .clone()
         .ok_or(ReceivedEventError::PendingLogEntry)?;
     let _transaction_index = entry
         .transaction_index
+        .clone()
+        .ok_or(ReceivedEventError::PendingLogEntry)?;
```

Comment:
Similarly, I think we can avoid cloning
```suggestion
    let _transaction_index = entry
        .transaction_index
        .as_ref()
        .ok_or(ReceivedEventError::PendingLogEntry)?;
```

Reply:
Good catch, thanks. See https://github.com/dfinity/ic/pull/5428/commits/0eb8d5cb8a09977b582d83554e3a939a2533b4a5